poly1305/src/soft.rs

```diff
@@ -0,0 +1,267 @@
+//! Software implementation of the Poly1305 state machine.
+
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//
+// This code originates from the rust-crypto project:
+// <https://github.com/DaGenix/rust-crypto>
+//
+// ...and was originally a port of Andrew Moons poly1305-donna
+// https://github.com/floodyberry/poly1305-donna
+
+use core::convert::TryInto;
+use universal_hash::generic_array::{typenum::U32, GenericArray};
+#[cfg(feature = "zeroize")]
+use zeroize::Zeroize;
+
+use crate::{Tag, BLOCK_SIZE};
+
+#[derive(Clone)]
+pub(crate) struct Poly1305State {
```

Comment:
I think I'd prefer `State` to `Poly1305State` in the spirit of [RFC#356](https://github.com/rust-lang/rfcs/blob/master/text/0356-no-module-prefixes.md) (although that deals specifically with modules, and not necessarily the name of the parent crate)

Reply:
Sounds good; I'll make this change in the next PR.